pkg/util/secret.go

```diff
@@ -27,7 +31,7 @@ func GetTargetSecret(client kubeclient.Interface, secretReferences []corev1.Obje
 		}
 	}
 
-	return nil, fmt.Errorf("no specific secret found in namespace: %s", targetNamespace)
+	return nil, errNotFound
```

Comment:
and if secretReferences is nil, variablel errNotFound will be nil too, and it may cause panics.

Reply:
Updated

Reply:
Yeah, the `errNotFound` now acts as a `NotFound` error, but the info in it is not correct.

How about this?
```diff
diff --git a/pkg/util/secret.go b/pkg/util/secret.go
index 8268071e..0652fab0 100644
--- a/pkg/util/secret.go
+++ b/pkg/util/secret.go
@@ -19,6 +19,9 @@ func GetTargetSecret(client kubeclient.Interface, secretReferences []corev1.Obje
                klog.V(2).Infof("checking secret: %s/%s", targetNamespace, objectReference.Name)
                secret, err := client.CoreV1().Secrets(targetNamespace).Get(context.TODO(), objectReference.Name, metav1.GetOptions{})
                if err != nil {
+                       if apierrors.IsNotFound(err) {
+                               continue
+                       }
                        return nil, err
                }

@@ -27,7 +30,12 @@ func GetTargetSecret(client kubeclient.Interface, secretReferences []corev1.Obje
                }
        }

-       return nil, fmt.Errorf("no specific secret found in namespace: %s", targetNamespace)
+       notfound := &apierrors.StatusError{ErrStatus: metav1.Status{
+               Message: fmt.Sprintf("no specific secret found"),
+               Reason:  metav1.StatusReasonNotFound,
+       }}
+
+       return nil, notfound
 }
```

Reply:
yes, if fixed like this, we'd better add a test to guarantee this StatusError is NotFound error, to avoid function `IsNotFound` implementation is modified.
```
&apierrors.StatusError{ErrStatus: metav1.Status{
+               Message: fmt.Sprintf("no specific secret found"),
+               Reason:  metav1.StatusReasonNotFound,
+       }}
```

Reply:
+1
You are right. But a little bit tricky.
Let's do it with this patch.